cloudsync/api_test.py

```diff
@@ -130,3 +150,172 @@ def test_process_transcode_results(mocker):
+                 ('MIT-6.046-2018-Fall-lec-mit-0000-2017apr06-0404.mp4', 'MIT-6.046-2018-Fall'),
+                 ('MIT-6.046-lec-mit-0000-2017apr06-0404.mp4', 'MIT-6.046'),
+                 ('MIT-6.046-lec-mit-0000-2017apr06-0404-L02.mp4', 'MIT-6.046-L02'),
+                 ('MIT-STS.001-2015-Fall-lec-mit-0000-2015dec21-2222-L03_2.mp4', 'MIT-STS.001-2015-Fall-L03'),
+                 ('MIT-HST.969-2018-Spring-lec-mit-0000-2018mar09-0001-L01.mp4', 'MIT-HST.969-2018-Spring-L01'),
+                 ('NUS-7.S939-2019-Spring-lec-mit-0000-2019apr02-1210-L05.mp4', 'NUS-7.S939-2019-Spring-L05'),
+                 ('MIT-Adhoc-lec-mit-0000-2017aug10-0954-2-190.mp4', 'MIT-Adhoc-2-190'),
+                 ('MIT-18.test-lec-mit-0000-2017aug10-0956-L01.mp4', 'MIT-18.test-L01'),
+                 ('MIT-18.test-lec-mit-0000-2017aug10-0956.mp4', 'MIT-18.test'),
+                 ('MIT-Simons Lecture Series-lec-mit-0000-2017aug10-0956-2-10.mp4', 'MIT-Simons Lecture Series-2-10'),
+                 ('/&*3:<>俺正和-lec-mit-0000-2017aug10-0956.mp4', '/&*3:<>俺正和'))
+    for (filename, title) in filenames:
+        assert extract_title(filename) == title
+
+
+def test_extract_bad_title():
+    """Assert that a VideoFilenameError is raised for a bad filename"""
+    bad_names = ('MIT-6.046-1510-L01.mp4',  # Missing -lec-mit-0000
+                 'Completely random name.mp4',  # No similarity to expected format
+                 'MIT-6.046-lec-mit-0000-2017apr06-L01')  # no extension
+
+    for name in bad_names:
+        with pytest.raises(VideoFilenameError):
+            extract_title(name)
+
+
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_watch_nouser():
+    """
+    Test that an exception is correctly handled when the LECTURE_CAPTURE_USER doesn't exist
+    """
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with pytest.raises(User.DoesNotExist):
+        process_watch_file(upload)
+    assert not Video.objects.filter(title=upload).exists()
+
+
+@mock_s3
+@override_settings(LECTURE_CAPTURE_USER='admin')
+def test_watch_s3_error():
+    """Test that an AWS S3 ClientError is correctly handled"""
+    UserFactory(username='admin')  # pylint: disable=unused-variable
+    s3 = boto3.resource('s3')
+    s3c = boto3.client('s3')
+    upload = 'MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4'
+    s3c.create_bucket(Bucket=settings.VIDEO_S3_WATCH_BUCKET)
+    bucket = s3.Bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    bucket.upload_fileobj(io.BytesIO(os.urandom(6250000)), upload)
+    with transaction.atomic():
+        with pytest.raises(ClientError) as exc:
+            process_watch_file(upload)
+            assert 'NoSuchBucket' in str(exc)
```

Comment:
Do you think it is necessary to assert this? I would prefer that an upgrade to their library changing error messages does not break our tests